src/Microsoft.Data.SqlClient/tests/ManualTests/ProviderAgnostic/ReaderTest/ReaderTest.cs

```diff
@@ -266,5 +267,118 @@ public static void TestMain()
+                {
+                    sqlCommand.CommandText = "select top 1 * from " + tableNameOne;
+                    DbDataReader reader = sqlCommand.ExecuteReader();
+
+                    reader.Read();
+
+                    Assert.True(reader[0].GetType() == typeof(int) && reader.GetFieldValue<int>(0) == 1);
+                    Assert.True(reader[1].GetType() == typeof(string) && reader.GetFieldValue<string>(1) == "Microsoft");
+                    Assert.True(reader[2].GetType() == typeof(bool) && reader.GetFieldValue<bool>(2) == true);
+                    Assert.True(reader[3].GetType() == typeof(short) && reader.GetFieldValue<short>(3) == 3274);
+                    Assert.True(reader[4].GetType() == typeof(byte) && reader.GetFieldValue<byte>(4) == 253);
+                    Assert.True(reader[5].GetType() == typeof(long) && reader.GetFieldValue<long>(5) == 922222222222);
+                   
+                    // Call Close when done reading.
+                    reader.Close();
+                }
+                using (SqlCommand sqlCommand = new SqlCommand("", con as SqlConnection))
+                {
+                    sqlCommand.CommandText = "select top 1 * from " + tableNameTwo;
+                    DbDataReader reader = sqlCommand.ExecuteReader();
+
+                    reader.Read();
+
+                    Assert.True(reader[0].GetType() == typeof(int) && reader.GetFieldValue<int>(0) == 1);
+                    Assert.True(reader[1].GetType() == typeof(double) && reader.GetFieldValue<double>(1) == 10.7);
+                    Assert.True(reader[2].GetType() == typeof(float) && reader.GetFieldValue<float>(2) == 123.546f);
+                    Assert.True(reader[3].GetType() == typeof(Guid) && reader.GetFieldValue<Guid>(3) == sqlguid.Value && reader.GetFieldValue<System.Data.SqlTypes.SqlGuid>(3).Value == sqlguid.Value);
+                    Assert.True(reader[4].GetType() == typeof(DateTime) && reader.GetFieldValue<DateTime>(4).ToString("dd/MM/yyyy HH:mm:ss") == dateTime.ToString("dd/MM/yyyy HH:mm:ss"));
+                    Assert.True(reader[5].GetType() == typeof(decimal) && reader.GetFieldValue<decimal>(5) == 280);
+                    Assert.True(reader[6].GetType() == typeof(DateTimeOffset) && reader.GetFieldValue<DateTimeOffset>(6) == dtoffset);
+                   
+                    // Call Close when done reading.
+                    reader.Close();
+                }
+
+                //cleanup
+                using (DbCommand cmd = provider.CreateCommand())
+                {
+                    cmd.Connection = con;
+                    cmd.CommandText = "drop table " + tableNameOne;
+                    cmd.ExecuteNonQuery();
+                }
+                using (DbCommand cmd = provider.CreateCommand())
+                {
+                    cmd.Connection = con;
+                    cmd.CommandText = "drop table " + tableNameTwo;
+                    cmd.ExecuteNonQuery();
+                }
```

Comment:
Use `try-catch` and include this section in the `finally` block to ensure it will be called.